### PR TITLE
fix: add select global filter

### PR DESCRIPTION
**Pull request by davidmenendez**
Contributes to #1893 

Screen Shot 2022-04-07 at 4 41 15 PM

adds global filter support. now a user can specify the properties to filter the list of items and those options should be generated based on the possible values.

```
globalFilters: [
      {
        id: 'fileType', // the property to look for in the entries
        label: 'File type', // label for the dropdown
      },
];
```
when `globalFilters` is detected an array of values will be generated `[{ id: 'fileType', opts: ['pdf', 'js', 'xml']}]` and then `opts` is used as the values in the dropdown.

### CODE REVIEW

File: packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                  selectedItem={getSelectedItem(filterOpts.id)}
+                  label={placeholder}
+                />
+              ))}
+            </div>
+          </div>
+          <ButtonSet className={`${blockClass}-button-set`}>
+            <Button
+              kind="secondary"
+              onClick={resetFilters}
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js:105]: The dropdown filters were reset when I tried this out, but the main content in the component did not get reset and I couldn't find a way to get the main content back.

Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js:105]: The dropdown filters were reset when I tried this out, but the main content in the component did not get reset and I couldn't find a way to get the main content back.

File: packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+          renderIcon={Filter16}
+          hasIconOnly
+          kind="ghost"
+          onClick={() => setOpen(!open)}
+          iconDescription={iconDescription}
+          className={`${blockClass}-toggle`}
+          size="md"
+        />
+      )}
+      {open && (
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js:83]: I think you should add the `useClickOutside` hook that we have here. I feel like the expected behavior here is that if you click outside the filter panel it should close.

Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js:83]: I think you should add the `useClickOutside` hook that we have here. I feel like the expected behavior here is that if you click outside the filter panel it should close.

File: packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js (hunk 1 above)
Comment (davidmenendez) [reply packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js:105]: i interpreted the reset functionality to act as like a only a reset for the filters. so you still have to click apply to apply the filters. but that's a good point. i'll double check with design on the UX 👍

File: packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js (hunk 2 above)
Comment (davidmenendez) [reply packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js:83]: i thought about that. i'll double check with design on that to verify that's the desired UX 👍

File: packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js (hunk 1 above)
Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js:105]: Ohh I see what you mean, ok.

File: packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js (hunk 2 above)
Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/AddSelect/AddSelectFilter.js:83]: Sounds good

**Review by matthewgallo — APPROVED** [review]
Since the things I commented on are UX related, I think we're good to approve 🎉

**Pull request merged**